File: FangZhiHu/CommonLib/src/main/java/dagger/module/CommonModule.java
package dagger.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import qj.ccmtjf.com.lib.HttpDataSource;
import qj.ccmtjf.com.lib.HttpDataSourceImpl;
import qj.ccmtjf.com.lib.LoanApp;

/**
 * Created by tongqinyuan on 2017/12/2.
 */
@Module
public class CommonModule {

    private final LoanApp mApp;

    public CommonModule(LoanApp app) {
        this.mApp = app;
    }

    @Singleton
    @Provides
    LoanApp provideLoanApp() {
        return mApp;
    }

    @Singleton
    @Provides
    HttpDataSource provideHttpDataSource(HttpDataSourceImpl httpDataSource) {
        return httpDataSource;
    }


}

File: FangZhiHu/CommonLibKotlin/src/main/java/com/ytq/dagger/module/CommonModule.kt
package com.ytq.dagger.module

import com.ytq.main.common.ApiMethodSource
import com.ytq.main.common.ApiMethodSourceImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by tongqinyuan on 2018/3/9.
 * 用于提供 数据接口的实现类
 */
@Module
class CommonModule {


    /**
     * 数据接口的实现类
     */
    @Singleton
    @Provides
    fun provideHttpDataSource(httpDataSource: ApiMethodSourceImpl): ApiMethodSource {
        return httpDataSource
    }


}